File: Scripts/volatile_and_force_compiler_task/regular-expressions/r/regular-expressions.r

executeTask <- function(i) {

pattern <- "string"
text1 <- paste("this is a matching string",i)
grep(pattern, c(text1))  # 1
gsub(pattern, "istyfied", c(text1))
	return(i)
}

r = 0

for(i in 0:10000000) {
	r = executeTask(i)
}

File: Scripts/volatile_and_force_compiler_task/regular-expressions/php/regular-expressions.php


<?php

$r = 1;

function executeTask($i) {

$string = 'this is a matching string' . $i;

if (preg_match('/string$/', $string))
{}
$string = preg_replace('/\bstring\b/', 'istyfied', $string);
	return $i;
}

for ( $i = 0; $i < 10000000; $i++) {
	$r = executeTask($i);
}
?>
